megamek/src/megamek/server/Server.java

```diff
@@ -6785,8 +6785,9 @@ else if (game.getBoard().onGround()) {
      *                   if it can't return)
      * @return
      */
-    private Vector<Report> processLeaveMap(Entity entity, int facing,
+    private Vector<Report> processLeaveMap(MovePath movePath,
                                            boolean flewOff, int returnable) {
+        Entity entity = movePath.getEntity();        
```

Comment:
I tested fleeing with a ground unit and got this:
Exception in thread "Packet Pump" java.lang.NullPointerException
        at megamek.server.Server.processLeaveMap(Server.java:6846)
        at megamek.server.Server.processMovement(Server.java:6938)
        at megamek.server.Server.receiveMovement(Server.java:5208)
        at megamek.server.Server.handle(Server.java:32004)
        at megamek.server.Server$PacketPump.run(Server.java:368)

It appears that when sending the flee command the movepath is generated without setting the Entity. I didn't look further than this, but it seems to me that the client should probably do that.

Reply:
The "processMovement" function already receives an entity from the client, so I just have it set the path's entity.